src/main/java/pl/allegro/tech/discovery/consul/recipes/watch/health/HealthServiceInstancesWatcher.java

```diff
@@ -48,13 +52,18 @@ public ServiceInstances decode(String s) {
             List<ServiceInstance> instances = services.stream()
                     .map(props -> {
                         Map<String, ?> service = requiredValue(props, "Service", Map.class);
-                        return new ServiceInstance(
-                                requiredValue(service, "ID", String.class),
-                                requiredValue(service, "Tags", List.class),
-                                requiredValue(service, "Address", String.class),
-                                requiredValue(service, "Port", Integer.class)
-                        );
-                    })
+                        try {
+                            return new ServiceInstance(
+                                    requiredValue(service, "ID", String.class),
+                                    requiredValue(service, "Tags", List.class),
+                                    requiredValue(service, "Address", String.class),
+                                    requiredValue(service, "Port", Integer.class)
+                            );
+                        } catch (Exception e) {
+                            logger.error(e.getMessage(), e.getCause());
```

Comment:
Czy ten log bedzie nas informowal też o jaką konkretnie instancje/service chodzi? Mysle, ze dobrze by bylo to miec w logach